src/main/java/jenkins/plugins/git/maintenance/MaintenanceUI.java

```diff
@@ -0,0 +1,167 @@
+package jenkins.plugins.git.maintenance;
+
+import antlr.ANTLRException;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.ManagementLink;
+import hudson.model.PeriodicWork;
+import hudson.security.Permission;
+import hudson.util.FormValidation;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+import org.kohsuke.stapler.verb.POST;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension
+public class MaintenanceUI extends ManagementLink {
+
+    private static final Logger LOGGER = Logger.getLogger(MaintenanceUI.class.getName());
+
+    @Override
+    public String getIconFileName() {
+        return jenkins.model.Jenkins.RESOURCE_PATH + "/plugin/git/icons/git-maintenance.svg";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Git Maintenance";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "maintenance";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Maintain your Repositories to improve git command performance.";
+    }
+
+    public @NonNull String getCategoryName() {
+        return "CONFIGURATION";
+    }
+
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public void doSave(StaplerRequest req, StaplerResponse res) throws IOException, ServletException {
+        if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+            LOGGER.log(Level.WARNING,"User doesn't have the required permission to access git-maintenance.");
+            res.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        JSONObject formData = req.getSubmittedForm();
+        MaintenanceTaskConfiguration config = GlobalConfiguration.all().get(MaintenanceTaskConfiguration.class);
+
+        // Need to check with mentors if I can use the below syntax.
+        assert config != null;
```

Comment:
use an if check and throw an exception if you have to instead of using an assert. To my knowledge, never use `assert` in a production environment. 
Read this for more context: https://stackoverflow.com/questions/2758224/what-does-the-java-assert-keyword-do-and-when-should-it-be-used